### ISSUE TITLE
Problem with VSCode and docker-php-xdebug

### ISSUE THREAD

**Issue by adrian-bobev** [mobtitude/docker-php-xdebug#1]
Hello,

I am trying to use the docker image with VSCode PHPDebug extension but unfortunatelly it does not work.

**docker-compose.yml file:**

> version: '3.8'
>
>   services:
>      wordpress:
>      container_name: wp
>      image: mobtitude/php-xdebug:7.2-apache
>      ports:
>         - "8080:80"
>      volumes: 
>         - "./src:/var/www/html/"

**launch.json file**
> {
>    "version": "0.2.0",
>    "configurations": [
>        {
>            "name": "Listen for XDebug",
>            "type": "php",
>            "request": "launch",
>            "port": 9000,
>            "pathMappings": {
>                "/var/www/html": "${workspaceRoot}/src"
>              }
>        }
>    ]
>}

And the code is in src folder. However, the PHPDebug does not stop on the break points

Any suggestions why?

Regards,
Adrian

**Comment by pszalko** [comment]
Hi @adrian-bobev,

I have no experience with VisualCode, however after a short research I found a plugin that maybe you use? I'm not sure if its default for VisualCode, but I checked docs for https://marketplace.visualstudio.com/items?itemName=felixfbecker.php-debug 

I see a few spots worth checking:

1. The `mobtitude/docker-php-xdebug` image starts Xdebug when the `XDEBUG_SESSION_START=session_name` parameter was passed in GET (or POST) HTTP. Maybe VisualCode does not pass this parameter?

2. The `mobtitude/docker-php-xdebug` image does not support XDebug autostart as documented in VisualCode plugin by default. If you want XDebug to always start debugger (not only when `XDEBUG_SESSION_START` parameter was passed), just add the option `xdebug.remote_autostart = 1` to the `xdebug.ini` file (see section xdebug config in readme in this repo how to do it). 

Let me know if that helped in any way.
